rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,212 @@
+import os
+import stat
+
+import rpm
+from rpmlint.checks.AbstractCheck import AbstractCheck
+from rpmlint.permissions import PermissionsParser, VariablesHandler
+
+
+class SUIDPermissionsCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.perms = {}
+
+        self.var_handler = VariablesHandler('/usr/share/permissions/variables.conf')
+
+        for fname in self._paths_to('permissions', 'permissions.secure'):
+            if not os.path.exists(fname):
+                continue
+
+            self._parse_profile(fname)
+
+    def _parse_profile(self, path):
+        parser = PermissionsParser(self.var_handler, path)
+        self.perms.update(parser.getEntries())
+
+    def _is_static_entry(self, entry):
+        # entries coming from the fixed permissions profile are considered
+        # static
+        return entry.profile.endswith('/permissions')
+
+    @staticmethod
+    def _paths_to(*file_names):
+        # we used to store the permissions data in /etc even though they aren't configuration files
+        # the whitelisting should check both paths (old /etc and new /usr/share) until all
+        # distributions using the old one (SLE15) are retired
+        for name in file_names:
+            # return the new path first.
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        need_set_permissions = False
+        found_suseconfig = False
```

Comment:
I think we can also drop this old `SuSEconfig` deprecation check. That tool was last used in SLE-11 if I remember correctly.